app/src/main/java/com/concordium/wallet/data/walletconnect/Params.kt

```diff
@@ -14,14 +14,18 @@ data class Params(
     var schema: Schema?
 ) : Serializable {
     fun parsePayload(): Payload? {
+        val gson = GsonBuilder()
+            .registerTypeAdapterFactory(NullableTypeAdapterFactory())
+            .create()
         return try {
-            val gson = GsonBuilder()
-                .registerTypeAdapterFactory(NullableTypeAdapterFactory())
-                .create()
-            gson.fromJson(payload, Payload::class.java)
-        } catch (ex: java.lang.Exception) {
-            Log.e(ex.toString())
-            null
+            gson.fromJson(payload, Payload.ComplexPayload::class.java)
```

Comment:
Can’t this be done looking at the payload instead of trying two times?

Reply:
payload is a string, I'm trying to parse it into an object here, this can throw an exception. i could theoretically do string matching here but that is extra complexity when changing/updating the payload model. i try to avoid string matching whenever possible.